src/accessibility/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { ElementHandle } from 'puppeteer';
+import { createClientRuntimeServer } from '../module-server/client-runtime-server';
+import { assertElementHandle } from '../utils';
+
+const accessibilityTreeSymbol: unique symbol = Symbol('PT Accessibility Tree');
+
+export interface AccessibilityTreeOptions {
+  /**
+   * Whether the accessible description of elements should be included in the tree.
+   * https://www.w3.org/TR/wai-aria-1.2/#dfn-accessible-description
+   * (default: true)
+   */
+  includeDescriptions?: boolean;
+  /**
+   * Whether to include text that is not part of an element's accessible name/description
+   * (default: false)
+   */
+  includeText?: boolean;
+}
+
+export const getAccessibilityTree = async (
+  element: ElementHandle,
+  options: AccessibilityTreeOptions = {},
+) => {
+  const serverPromise = createClientRuntimeServer();
+  assertElementHandle(element, getAccessibilityTree);
+
+  const { port } = await serverPromise;
+
+  const result: string = await element.evaluate(
+    // Using new Function to avoid babel transpiling the import
+    // @ts-expect-error pptr's types don't like new Function
+    new Function(
+      'element',
+      'options',
+      `return import("http://localhost:${port}/@pleasantest/accessibility")
+        .then(accessibility => accessibility.getAccessibilityTree(element, options))`,
+    ),
+    options,
+  );
+
+  return {
+    [accessibilityTreeSymbol]: result,
+    toString: () => result,
+  };
+};
+
+expect.addSnapshotSerializer({
```

Comment:
Can you help me understand what’s going on here? LOL

Reply:
Jest's inline snapshots have extra quotes for strings:

```js
expect("hello").toMatchInlineSnapshot(`
  "hello"
`)
```

I didn't like the extra quotes for the accessibility tree (it would also escape quotes inside). You can tell Jest how to print certain kinds of objects: https://jestjs.io/docs/expect#expectaddsnapshotserializerserializer

So I made `getAccessibilityTree` return an object and I used `addSnapshotSerializer` to tell Jest how to print the object (without quotes on the outside)

Reply:
Got it. Can you add a brief comment for future devs? Thank you. 🙂